### ISSUE TITLE
Syntax errors are not displayed properly

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml-ppx/ppxlib#221]
Using the following test-case
```
$ cat dune
(executable
 (name test)
 (preprocess (pps ppxlib.metaquot)))
$ cat test.ml
let (let+) _ _ = Obj.magic ()
let (and+) _ _ = Obj.magic ()

let () =
  let+ x = ""
  let+ y = ""
  and+ z = ""
  in
  ()
```
and calling dune will result in the following, rather unhelpful, error message:
```
$ dune build --verbose
(cd _build/default && .ppx/0224ad3443a846e54f1637fccb074e7d/ppx.exe -o test.pp.ml --impl test.ml -corrected-suffix .ppx-corrected -diff-cmd - -dump-ast)
File "test.ml", line 1:
Error: Syntax error
```
in contrast, calling the compiler directly results in a more helpful error message:
```
$ ocamlopt test.ml
File "test.ml", line 6, characters 2-6:
6 |   let+ y = ""
      ^^^^
Error: Syntax error
```
----------
Test setup:
* `ppxlib` version: `0.22.0`
* `dune` version: `2.8.2`

**Comment by jeremiedimino** [comment]
BTW, the first weird thing here is that the location is different:

- line 1 in the first case
- line 6, characters 2-6 in the second

maybe the non-printing of the file excerpt is due to the location being wrong. The fact that is has no characters information seems odd.

**Comment by Armael** [comment]
This is most likely the symptom of a `Location.none` leaking out. In that case you still get the right filename because the printing code [will default to !Location.input_name](https://github.com/ocaml/ocaml/blob/trunk/parsing/location.ml#L190), but then you get no information (except `line 1` [for backward compatibility reasons](https://github.com/ocaml/ocaml/blob/trunk/parsing/location.ml#L212)).

**Comment by aantron** [comment]
`git bisect` indicates that the bug was introduced in a50690c5f54ed9b89f6a16906f1b08155db1fe32.

Could you take a look? This is a major usability issue at the moment.

**Comment by aantron** [comment]
A workaround in the meantime is to use ppxlib 0.20.0.

I have at least one project (Bisect_ppx) which requires a higher version, though.

**Comment by pitag-ha** [comment]
This should be fixed on master now. I'll let you all know when making the next release.

**Comment by aantron** [comment]
Thank you!

**Comment by pitag-ha** [comment]
> it looks like the code preview which is printed when the compiler detects an error isn't printed by ppxlib.

Do you mean the
```
6 |   let+ y = ""
      ^^^^
```
or similar? That is usually printed but [muted in the test](https://github.com/ocaml-ppx/ppxlib/pull/247/files#diff-1529cf5e2b70f7d642670ef6c302fe7e6504bedaec3739698642105bbeb55c98R4). The reason for muting it is that it is only printed from a certain compiler version on (IIRC from 4.06 on, but I'm not sure) and I need the test to be version agnostic.

**Comment by kit-ty-kate** [comment]
oh i see! Thanks a lot!

**Comment by pitag-ha** [comment]
This should be fixed in 0.22.1 now (and also in 0.21.1).
